docs/en/Community-Articles/2024-05-09-what-is-object-to-object-mapping/post.md

```diff
@@ -0,0 +1,122 @@
+            FirstName = input.FirstName,
+            LastName = input.LastName,
+            // ...other
+        };
+        
+        return customerDto;
+    }
+
+````
+This code is repetitive, tedious to write. It violates DRY principles, makes your code very complex and reduces readability.  Instead of using this approach, you can use the automapper which does the matching automatically. Here is a better code:
+
+
+````csharp
+        public virtual async Task<CustomerDto> CreateAsync(CustomerCreateDto input)
+        {
+
+            var customer = await _customerManager.CreateAsync(
+            input.BirthDay, input.MembershipDate, input.FirstName, input.LastName
+            );
+
+            return ObjectMapper.Map<Customer, CustomerDto>(customer);
+        }
+````
+
+# What is AutoMapper?
+
+![Swagger](./images/automapper.png)
+
+Automapper is a .net library that automates object to object mapping. The implementation of Automapper is quite simple. In this chapter I will show you how to use the Automapper library in an ABP-based application. For this reason, I assume that you already have an ABP-based application created. If you have not yet created an ABP-based application, please follow the [Getting Started documentation](https://docs.abp.io/en/abp/latest/Getting-Started-Create-Solution?UI=MVC&DB=EF&Tiered=No).
+
+Create a domain entity similar to this one: 
+
+````csharp
+    public class Customer : FullAuditedAggregateRoot<Guid>
+    {
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime BirthDay { get; set; }
+        public DateTime MembershipDate { get; set; }
+    }
+````
+`Custumer` contains a lot of attributes. we don't want to give all of these attributes to users, so create dto with only the necessary attributes: 
+
+````csharp
+    public class CustomerGetDto
+    {
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime BirthDay { get; set; }
+    }
+````
+
+You can use this code to list customers:
+
+````csharp
+public virtual async Task<PagedResultDto<CustomerGetDto>> GetListAsync(GetCustomersInput input)
+ {
+     var totalCount = await _customerRepository.GetCountAsync(input.FilterText, input.FirstName, input.LastName, input.BirthDayMin, input.BirthDayMax, input.MembershipDateMin, input.MembershipDateMax);
+     var items = await _customerRepository.GetListAsync(input.FilterText, input.FirstName, input.LastName, input.BirthDayMin, input.BirthDayMax, input.MembershipDateMin, input.MembershipDateMax, input.Sorting, input.MaxResultCount, input.SkipCount);
+
+     return new PagedResultDto<CustomerGetDto>
+     {
+         TotalCount = totalCount,
+         Items = ObjectMapper.Map<List<Customer>, List<CustomerGetDto>>(items)
+     };
+ }
+````
+
+Update your `YourApplicationAutoMapperProfile` class so that `Customer` and `CustomerGetDto` entities can convert to each other.
```

Comment:
Instead of using **convert** as the term, it would be better to use **map** as the word:

```suggestion
After mapping the `Customer` entity to the `CustomerGetDto` class. Before running our application, we should define the mappings in the `*AutoMapperProfile` class in the `*.Application` project as follows:
```